### PR TITLE
refactor: Streamline message sending process

### PR DESCRIPTION
**Pull request by atomrc**
Improve the concerns of the different components of the inputBar:

- the sending button doesn't know anything about the rich text editor anymore
- the rich text editor is responsible for transforming the mentions directly (no leak of how it's implemented)

### CODE REVIEW

File: src/script/components/InputBar/InputBar.tsx [hunk 1]
```diff
@@ -193,11 +194,9 @@ export const InputBar = ({
     setCurrentMentions([]);
 
     if (resetInputValue) {
-      lexicalRef.current?.update(() => {
+      editorRef.current?.update(() => {
         $getRoot().clear();
       });
-
-      setInputValue('');
```
Comment (atomrc) [inline src/script/components/InputBar/InputBar.tsx:200]: We never need to manually update the state anymore, it's automatically updated from lexical's update callback

File: src/script/components/LexicalInput/components/SendMessageButton/SendMessageButton.tsx [hunk 2]
```diff
@@ -1,119 +0,0 @@
...
-
-        handleSendMessage();
-
-        return false;
-      },
-      COMMAND_PRIORITY_LOW,
-    );
-
-    return () => {
-      removeListener();
```
Comment (atomrc) [inline src/script/components/LexicalInput/components/SendMessageButton/SendMessageButton.tsx:102]: This entire logic has been moved to components that should know about the editor (the sending button should not)

**Review by V-Gira — COMMENTED** [review]
I double-checked that we're not reintroducing that bug https://wearezeta.atlassian.net/browse/WPB-3070
(sending a populated mention array after typing a mention and deleting it)
All good :+1:

File: src/script/components/InputBar/InputBar.tsx [hunk 3]
```diff
@@ -395,13 +394,16 @@ export const InputBar = ({
...
     } else {
-      sendMessage(text, mentions);
+      sendTextMessage(text, mentions);
     }
 
-    lexicalRef.current?.focus();
+    editorRef.current?.focus();
+    editorRef.current?.update(() => {
+      $getRoot().clear();
+    });
```
Comment (przemvs) [inline src/script/components/InputBar/InputBar.tsx]: This one doesn't work ? 🤔 

https://lexical.dev/docs/api/classes/lexical.LexicalEditor#focus

```suggestion
    editorRef.current?.focus(() => {
      $getRoot().clear();
    });
```

Comment (atomrc) [reply src/script/components/InputBar/InputBar.tsx]: oh, good call, let me try 👍

Comment (atomrc) [reply src/script/components/InputBar/InputBar.tsx]: Nope, this doesn't work, you need to call `update` in order to be able to update the editor

```
Lexical.dev.js:8973 Uncaught Error: Unable to find an active editor state. State helpers or node methods can only be used synchronously during the callback of editor.update() or editorState.read().
```

File: src/script/components/InputBar/InputBar.tsx [hunk 4]
```diff
@@ -181,7 +182,7 @@ export const InputBar = ({
   // To be changed when design chooses a breakpoint, the conditional can be integrated to the ui-kit directly
   const isScaledDown = useMatchMedia('max-width: 768px');
 
-  const showGiphyButton = inputValue.length > 0 && inputValue.length <= config.GIPHY_TEXT_LENGTH;
+  const showGiphyButton = messageContent.text.length > 0 && messageContent.text.length <= config.GIPHY_TEXT_LENGTH;
```
Comment (przemvs) [inline src/script/components/InputBar/InputBar.tsx:186]: Please create variable messageLength or smth like that

File: src/script/components/InputBar/InputBar.tsx [hunk 5]
```diff
@@ -307,23 +305,23 @@ export const InputBar = ({
       return () => {};
     }
 
-    if (inputValue.length > 0) {
+    if (messageContent.text.length > 0) {
```
Comment (przemvs) [inline src/script/components/InputBar/InputBar.tsx:309]: You can reuse this variable here.

**Review by przemvs — APPROVED** [review]

**Pull request merged**